src/GameState/getMarblesCoordinateToAnimate.tsx

```diff
@@ -0,0 +1,35 @@
+import { Graph } from '../interface';
+
+export const getMarblesCoordinateToAnimate = (
+  marbleCoordinate: any,
+  graph: Graph | null | undefined,
+  direction: string,
+) => {
+  if (!marbleCoordinate || !graph || !direction) {
+    return [];
+  }
+
+  let currentNode = graph.nodes[`${marbleCoordinate.x},${marbleCoordinate.y}`];
+
+  const nodes = [currentNode];
+  while (true) {
```

Comment:
It is easier to read here when the loop is over. With this, you have to read the whole content of the loop to know when to leave it. So change your loop construction to have a real breaking condition.